scrapy/utils/conf.py

```diff
@@ -34,6 +34,16 @@ def _map_keys(compdict):
             _check_components(compdict)
             return {convert(k): v for k, v in six.iteritems(compdict)}
 
+    def _ensure_numeric_value(compdict):
+        for name, value in six.iteritems(compdict):
+            if value is None or isinstance(value, (int, float)):
+                continue
+            try:
+                compdict[name] = float(value)
```

Comment:
One more question: why is converting values to floats needed? Shouldn't we just raise an exception if the value is not int/float, what's the point of supporting non-numeric values?